examples/enricher/src/test/java/org/apache/apex/examples/enricher/ApplicationTest.java

```diff
@@ -36,16 +44,65 @@
   @Test
   public void testApplication() throws Exception
   {
-    try {
-      LocalMode lma = LocalMode.newInstance();
-      Configuration conf = new Configuration(false);
-      conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties.xml"));
-      lma.prepareDAG(new EnricherAppWithJSONFile(), conf);
-      LocalMode.Controller lc = lma.getController();
-      lc.run(10000); // runs for 10 seconds and quits
-    } catch (ConstraintViolationException e) {
-      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    Configuration conf = new Configuration(false);
+    conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties.xml"));
+    EmbeddedAppLauncher<?> launcher = Launcher.getLauncher(Launcher.LaunchMode.EMBEDDED);
+    Attribute.AttributeMap launchAttributes = new Attribute.AttributeMap.DefaultAttributeMap();
+    launchAttributes.put(EmbeddedAppLauncher.RUN_ASYNC, true);
+    EnricherAppWithJSONFile enricherAppWithJSONFile = new EnricherAppWithJSONFile();
+    Launcher.AppHandle appHandle = launcher.launchApp(enricherAppWithJSONFile, conf, launchAttributes);
+    File folder = new File("target/enrichedData");
+    File[] listOfFiles = null;
+    // First ensure that the output file is generated and the expected lines match
+    // The test main thread sleeps until the expected lines are generated in the output file
+    while (true) {
```

Comment:
should have a timeout

Reply:
Added a timeout and refactored the code a bit as the test function was turning out to be too long.